packages/components/src/components/file/file.component.ts

```diff
@@ -331,10 +382,20 @@ export default class SynFile extends SynergyElement implements SynergyFormContro
     e.preventDefault();
     e.stopPropagation();
 
-    const files = e.dataTransfer?.files;
+    if (!e.dataTransfer) {
+      return;
+    }
+
+    let { files } = e.dataTransfer;
```

Comment:
This potentially overrides the original dataTransfer. Maybe something like this would be possible, too?


```suggestion
    const files = this.webkitdirectory
      ? await this.handleTransferItems(e.dataTransfer?.items)
      : e.dataTransfer.files;
```